src/main/java/pageobject/AbstractPage.java

```diff
@@ -0,0 +1,27 @@
+package pageobject;
+
+import core.IWebDriverProvider;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import java.time.Duration;
+
+public abstract class AbstractPage {
+    IWebDriverProvider driver;
+
+    public AbstractPage(IWebDriverProvider driver) {
+        this.driver = driver;
+    }
+
+    public void openWindow(String url) {
+        driver.getWebDriver().get(url);
+        driver.maximizeWindow();
+    }
+
+    public boolean waitUntilElementVisible(WebElement element) {
+        WebDriverWait wait = new WebDriverWait(driver.getWebDriver(), Duration.ofSeconds(20));
+        wait.until(ExpectedConditions.visibilityOf(element));
+        return element.isDisplayed();
+    }
```

Comment:
Waiting for element to be displayed and then checking again that it's displayed doesn't make sense because if element is not there, wait will fail with exception. So your method has 2 options either return true or fail with exception, it won't return false. 
Therefore implementations should be changed as well as usages